### PR TITLE
Add sort parameters to URL for Violations page

### PR DESCRIPTION
**Pull request by dvail**
## Description

This allows the Violations page and Observed CVEs component to track "sort" as part of the URL parameters. The useURLSort and changes to the Observed CVEs component were cherry picked from @sachaudh's PR #470, with minor changes to preserve referential equality across renders.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added

## Testing Performed

Visiting Violations page with no parameters defaults sort to most recent:
image

Clicking on Severity sorts list by severity, descending, and tracks params in URL:
image

Clicking on Severity when sorting descended reverses the sort order:
image

Tested that a page refresh or direct URL visit with sorting parameters will apply the sort options on the initial render.

Tested that the URL sort parameters coexists with existing search and pagination parameters (sorted by time):
image

Tested that sorting works identically on the Observerd CVEs component (Vuln Mgmt -> Images -> Image Detail View):
image
image

### CODE REVIEW

**Review by sachaudh — COMMENTED** [review]
LGTM but I think something I forgot to mention was that the Observed, Deferred, False Positive tabs don't append another value to the URL (ie. like `.../observed`) so having the sort fields be URL-based would probably not make as much sense in that situation. We talked about it a while ago and decided to make the pagination, sorting, and filtering for that page be through the component state rather than URL-state. I think what you can do is probably just omit the ObservedCVEs change.

https://user-images.githubusercontent.com/4805485/163025349-dd0471ce-7fd3-4067-a2df-6b2c16504778.mov

**Comment by dvail** [comment]
Got it, good call. I've reverted the changes to the ObservedCVE components.

**Review by sachaudh — APPROVED** [review]
Nice, thanks for the change! Looks good to me

File: ui/apps/platform/src/Containers/Violations/ViolationsTablePanel.tsx [hunk 1]
```diff
@@ -13,13 +13,14 @@ import {
 import { TableComposable, Thead, Tbody, Tr, Th, Td } from '@patternfly/react-table';
 
 import useTableSelection from 'hooks/useTableSelection';
-import { TableColumn, SortDirection } from 'hooks/useTableSort';
+import { TableColumn } from 'hooks/useTableSort';
```
Comment (alwayshooin) [inline ui/apps/platform/src/Containers/Violations/ViolationsTablePanel.tsx:16]: we can probably pull this out into a separate table types file

Comment (dvail) [reply ui/apps/platform/src/Containers/Violations/ViolationsTablePanel.tsx:16]: Looking through some related areas I found some other stuff that could be moved to a table types file, I'll open up a follow up PR for moving those over.

**Pull request merged**